src/agent/onefuzz-supervisor/src/worker.rs

```diff
@@ -228,8 +228,6 @@ impl IWorkerRunner for WorkerRunner {
         cmd.arg("managed");
         cmd.arg("config.json");
         cmd.arg(setup_dir);
-        cmd.stderr(Stdio::piped());
-        cmd.stdout(Stdio::piped());
```

Comment:
This isn't a viable fix.

We capture stdout & stderr from the agent to report to the user why something went wrong.  The stdout & stderr is reported back to the service on task failure.  Right now, the service only saves the last 4k of stdout and 4k of stderr.

Reply:
An *extremely* naive impl:

```rust
async fn last_4k_string(stream: impl AsyncRead + Unpin) -> Result<String> {
    let mut result = String::new();

    let mut stream = BufReader::new(stream);
    loop {
        let mut buf = vec![];
        let bytes_read = stream
            .read_until(b'\n', &mut buf)
            .await
            .map_err(|e| format_err!("unable to read stream {}", e))?;
        if bytes_read == 0 {
            break;
        }
        let line = String::from_utf8_lossy(&buf).to_string();
        result.push_str(&line);
        result.truncate(4096);
    }
    Ok(result)
}


pub async fn monitor_process_last_4k(mut process: Child) -> Result<Output> {
    let stdout = process
        .stdout
        .take()
        .ok_or_else(|| format_err!("stdout not captured"))?;
    let stderr = process
        .stderr
        .take()
        .ok_or_else(|| format_err!("stderr not captured"))?;

    let stdout_monitor = last_4k_string(stdout);
    let stderr_monitor = last_4k_string(stderr);

    // even if stdout/stderr read fails, wait on the others.
    let (stdout_result, stderr_result, process_result) =
        futures::join!(stdout_monitor, stderr_monitor, process);
    let exit_status = process_result?.into();

    let stdout = stdout_result.unwrap_or_else(|x| format!("unable to read stdout: {}", x));
    let stderr = stderr_result.unwrap_or_else(|x| format!("unable to read stderr: {}", x));

    Ok(Output {
        stdout,
        stderr,
        exit_status,
    })
}
```

Reply:
Ah, right, last night I just wasn't seeing that we were still doing that!

In that case I agree, something like a 4kb tail is the least viable solution. Will update this PR accordingly.